common/errors.go

```diff
@@ -1945,8 +1967,39 @@ func HasErrorCode(err error, codes ...ErrorCode) bool {
 	return false
 }
 
+func IsRetryableTowardNetwork(err error) bool {
+	// Missing data errors -> Retry
+	if HasErrorCode(err, ErrCodeEndpointMissingData) {
+		return true
+	}
+
+	// If the error says it's explicitly retryable/not retryable towards network
+	if se, ok := err.(StandardError); ok {
+		if rt, ok := se.DeepSearch("retryableTowardNetwork").(bool); ok {
+			return rt
+		}
+	}
+
+	// For exhausted upstreams, check if any of the underlying errors are retryable towards upstream
+	if HasErrorCode(err, ErrCodeUpstreamsExhausted) {
+		if exher, ok := err.(*ErrUpstreamsExhausted); ok {
+			errs := exher.Errors()
+			if len(errs) > 0 {
+				for _, e := range errs {
+					if IsRetryableTowardsUpstream(e) {
+						return true
+					}
+				}
+			}
+		}
+	}
+
+	// If the error is retryable towards upstream, then it is also retryable towards network
+	return IsRetryableTowardsUpstream(err)
```

Comment:
If this is true, then does it mean we're doing a redundant `IsRetryableTowardsUpstream` in failsafe.go, for the flow of network scope? Is there any chance to simplify that?